app/main.py

```diff
@@ -1 +1,48 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(BackendDeveloper, FrontendDeveloper):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django", "JavaScript", "HTML", "CSS"]
+
```

Comment:
you hard code skills for FullStackDeveloper. But what if BackendDeveloper and FrontendDeveloper will have another skills? SoftwareEngineer include learn_skill method  so child classes  instances could have more skills. you need to extend list of skills in BackendDeveloper and FrontendDeveloper classes. Pytests for this tack includes such a cases. Did you have all tests run successfully   ?

Reply:
Yes, that worked. You may check it by yourself in pycharm. Nevertheless, I have updated the code according to your comment.

> you hard code skills for FullStackDeveloper. But what if BackendDeveloper and FrontendDeveloper will have another skills? SoftwareEngineer include learn_skill method so child classes instances could have more skills. you need to extend list of skills in BackendDeveloper and FrontendDeveloper classes. Pytests for this tack includes such a cases. Did you have all tests run successfully ?